### PR TITLE
#197 Insert PaymentFields on checkout page

### PR DESCRIPTION
**Pull request by eugene-shab**
Feat: Implement `PaymentFields` on checkout page

Refs: #197

### CODE REVIEW

**Review by garthbrantley — CHANGES_REQUESTED** [review]
See comments below. I think there is a still a lot of work to do.

File: view/base/requirejs-config.js [hunk 1]
```diff
@@ -2,7 +2,8 @@
 var config = {
     map: {
         '*': {
-            spreedly: 'https://core.spreedly.com/iframe/iframe-v1.min.js'
+            spreedly: 'https://core.spreedly.com/iframe/iframe-v1.min.js',
```
[the remark this answers is not in the document]
Comment (eugene-shab) [reply view/base/requirejs-config.js:5]: Keep it until update MyAccount section.

File: Gateway/Response/ThreeDSecure.php [hunk 2]
```diff
@@ -0,0 +1,54 @@
+<?php
```
[the remark this answers is not in the document]
Comment (eugene-shab) [reply Gateway/Response/ThreeDSecure.php]: removed

File: view/frontend/web/js/model/payment/cc-form.js [hunk 3]
```diff
@@ -45,12 +45,17 @@ define(
...
+                        'creditCardFirstDigits',
+                        'creditCardLastDigits',
                         'paymentMethodToken',
-                        'selectedCardType'
+                        'selectedCardType',
                     ]);
 
                 return this;
             },
+            getThreeDSBrowserInfo: function () {
```
[the remark this answers is not in the document]
Comment (eugene-shab) [reply view/frontend/web/js/model/payment/cc-form.js]: removed

File: view/frontend/web/js/view/payment/method-renderer/cc-form.js [hunk 4]
```diff
@@ -28,37 +29,108 @@ define(
...
+                        'creditcard_last_digits': this.creditCardLastDigits(),
+                        'creditcard_type': this.selectedCardType(),
+                    },
                 };
+
+                data.additional_data.browser_info = this.getThreeDSBrowserInfo();
+                return data;
+            },
+
+            getThreeDSBrowserInfo: function () {
```
[the remark this answers is not in the document]
Comment (eugene-shab) [reply view/frontend/web/js/view/payment/method-renderer/cc-form.js]: removed

File: Platform/Tool/Oauth.php [hunk 5]
```diff
@@ -28,4 +28,16 @@ public function getWidgetAccessTokenByGuest($websiteId = null)
             'scope' => 'widget',
         ]);
     }
+
+    /**
+     * @param string|int|null $websiteId
+     * @return mixed
+     */
+    public function getPaymentFieldAccessToken($websiteId = null): mixed
```
[the remark this answers is not in the document]
Comment (eugene-shab) [reply Platform/Tool/Oauth.php]: renamed

**Review by garthbrantley — APPROVED** [review]
Code reviewed and approved. Still needs testing.

**Comment by garthbrantley** [comment]
Closing in favor of #211 and #212

**Pull request closed without merging**